### PR TITLE
Velodyne pcl

### PR DESCRIPTION
**Pull request by spuetz**
Regrading #287 The `velodyne_pcl` ROS package restores the conversion file `point_types.h`.
The README explains how to use it.

The following has been done:
- Fixed assignment of the time value in the organized point cloud container
- Added `velodyne_pcl` ROS package with the `point_types.h` header file.
- Added a RAEDME.md file to explain how to convert velodyne ROS point clouds to PCL
- Renamed both containers to cover the previously added time property.

### CODE REVIEW

**Comment by spuetz** [comment]
@JWhitleyWork Could you test the conversion? Cheers.

**Comment by JWhitleyWork** [comment]
@spuetz Just two requests if you could: 

1. Please add a nodelet which simply does the conversion and a launch file that runs the nodelet with parameters for a nodelet manager and input and output topics. 
2. Please add `roslint` to the package and the associated test.

**Comment by JWhitleyWork** [comment]
Thanks again for working on this.

**Comment by spuetz** [comment]
The conversion is also automatically done by the subscriber and the publisher. See http://wiki.ros.org/pcl/Overview#Publishing_and_subscribing_to_point_cloud_messages
I think it would make sense to just provide a test where we know which data is in the could and then compare it after conversion.
Would be good if someone could write the test, I don't have much time... and I think you are more into the testing of your laser scan data with the pcap data.

**Comment by JWhitleyWork** [comment]
Thanks for the clarification, @spuetz.

**Comment by JWhitleyWork** [comment]
@spuetz I'll add `roslint` to the package.  However, I've also just realized that this MR breaks API in another way by doing away with the `PointCloudXYZIR` type. While I agree that the type name should match the contents, this name change breaks existing use of the type by [many, many](https://github.com/search?q=velodyne_pointcloud+PointXYZIR&type=Code) examples on Github. This change can be introduced in Noetic, but not in the Melodic LTS. Please revert this commit.

**Comment by spuetz** [comment]
Someone introduced the time field in that container, instead of creating a new one. That mainly caused the problem. Cleanest solution would be to add another container without the time field and to add the corresponding point type here.

These depending packages, you mentioned, they also need to change the include file to `velodyne_pcl/point_types.h`. Additionally they need to install the `velodyne_pcl` package.

**Comment by spuetz** [comment]
@JWhitleyWork  Any news here? See my comment above.

**Comment by JWhitleyWork** [comment]
@spuetz Sorry I haven't responded lately. Yes, I understand that users will need to change the include path and depend on `velodyne_pcl`. Since these changes are necessary already, changing the point type doesn't seem like that big of a deal.

I'm creating a PR to your branch right now to add `roslint`.

**Comment by JWhitleyWork** [comment]
PR Created for `roslint`: https://github.com/spuetz/velodyne/pull/1

**Comment by YoshuaNava** [comment]
@JWhitleyWork @spuetz I'm looking forward to this PR being merged and would be glad to help test if needed. I have a VLP-16 that I can use for tests.

**Comment by JWhitleyWork** [comment]
@YoshuaNava I would really appreciate that! Could you possibly test this prior to merge? If you clone from @spuetz fork of the repository and check out his `velodyne_pcl` branch, you should be able to build and test it. Please let me know how it goes and I'll merge it immediately after!

**Comment by YoshuaNava** [comment]
@JWhitleyWork Can you send me a short list of the functionality I should text, and what is expected from each test?

File: velodyne_pcl/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 2.8.3)
```
Comment (JWhitleyWork) [inline velodyne_pcl/CMakeLists.txt:1]: Since this is now targeting Noetic, please update to 3.5.

File: velodyne_pcl/package.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<package format="2">
```
Comment (JWhitleyWork) [inline velodyne_pcl/package.xml:2]: Please update to Format 3 and include the XML namespace declaration. e.g.

```
<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
```

File: velodyne_pcl/package.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  <name>velodyne_pcl</name>
+  <version>0.0.0</version>
```
Comment (JWhitleyWork) [inline velodyne_pcl/package.xml:4]: Please set to version 1.5.2 to match existing version for ease of upgrade with `catkin` tools.

File: velodyne_pcl/package.xml (hunk 2 above)
Comment (JWhitleyWork) [reply velodyne_pcl/package.xml:2]: Thanks for adding the URL. However, you need to change the `<package>` tag to match (format="3").

**Comment by JWhitleyWork** [comment]
Last item: can you rebase on master so it uses Noetic CI?

**Pull request merged**